packages/wasm/crate/src/keys.rs

```diff
@@ -44,122 +45,85 @@ pub fn load_proving_key(parameters: JsValue, key_type: &str) -> WasmResult<()> {
 /// generate a spend key from a seed phrase
 /// Arguments:
 ///     seed_phrase: `string`
-/// Returns: `bech32 string`
+/// Returns: `Uint8Array representing inner SpendKey`
 #[wasm_bindgen]
-pub fn generate_spend_key(seed_phrase: &str) -> WasmResult<JsValue> {
+pub fn generate_spend_key(seed_phrase: &str) -> WasmResult<Vec<u8>> {
     utils::set_panic_hook();
 
     let seed = SeedPhrase::from_str(seed_phrase)?;
     let path = Bip44Path::new(0);
     let spend_key = SpendKey::from_seed_phrase_bip44(seed, &path);
-
-    let proto = spend_key.to_proto();
-
-    let spend_key_str = bech32str::encode(
-        &proto.inner,
-        bech32str::spend_key::BECH32_PREFIX,
-        bech32str::Bech32m,
-    );
-
-    Ok(JsValue::from_str(&spend_key_str))
+    Ok(spend_key.encode_to_vec())
 }
 
 /// get full viewing key from spend key
 /// Arguments:
-///     spend_key_str: `bech32 string`
-/// Returns: `bech32 string`
+///     spend_key: `byte representation inner SpendKey`
+/// Returns: `Uint8Array representing inner FullViewingKey`
 #[wasm_bindgen]
-pub fn get_full_viewing_key(spend_key: &str) -> WasmResult<JsValue> {
+pub fn get_full_viewing_key(spend_key: &[u8]) -> WasmResult<Vec<u8>> {
     utils::set_panic_hook();
 
-    let spend_key = SpendKey::from_str(spend_key)?;
-
+    let spend_key: SpendKey = SpendKey::decode(spend_key)?;
     let fvk: &FullViewingKey = spend_key.full_viewing_key();
-
-    let proto = fvk.to_proto();
-
-    let fvk_bech32 = bech32str::encode(
-        &proto.inner,
-        bech32str::full_viewing_key::BECH32_PREFIX,
-        bech32str::Bech32m,
-    );
-    Ok(JsValue::from_str(&fvk_bech32))
+    Ok(fvk.encode_to_vec())
 }
 
 /// Wallet id: the hash of a full viewing key, used as an account identifier
 /// Arguments:
-///     full_viewing_key: `bech32 string`
-/// Returns: `bech32 string`
+///     full_viewing_key: `byte representation inner FullViewingKey`
+/// Returns: `WalletId`
 #[wasm_bindgen]
-pub fn get_wallet_id(full_viewing_key: &str) -> WasmResult<String> {
+pub fn get_wallet_id(full_viewing_key: &[u8]) -> WasmResult<JsValue> {
     utils::set_panic_hook();
 
-    let fvk = FullViewingKey::from_str(full_viewing_key)?;
-    Ok(fvk.wallet_id().to_string())
+    let fvk: FullViewingKey = FullViewingKey::decode(full_viewing_key)?;
+    let result = serde_wasm_bindgen::to_value(&fvk.wallet_id())?;
+    Ok(result)
```

Comment:
@grod220 
`WalletId` has no method `.encode_to_vec()`

Reply:
Looks like WalletId is missing a trait in the core repo. I'll open a PR to fix there. In the meantime, let's do:

```rust
pub fn get_wallet_id(full_viewing_key: JsValue) -> WasmResult<Vec<u8>> {
    utils::set_panic_hook();

    let fvk: FullViewingKey = serde_wasm_bindgen::from_value(full_viewing_key)?;
    // Can do `fvk.wallet_id().encode_to_vec()` when Domain impl added to WalletId in core
    let wallet_id_proto = WalletId::from(fvk.wallet_id());
    Ok(wallet_id_proto.encode_to_vec())
}
```

Reply:
`penumbra_proto::core::keys::v1::WalletId` is also missing a trait